packages/machine-extractor/src/extractAction.ts

```diff
@@ -0,0 +1,411 @@
+    const arg2 = node.arguments[1];
+    const arg3 = node.arguments[2];
+
+    // Actor
+    // sendTo('actorName')
+    if (t.isStringLiteral(arg1)) {
+      actorRef = { type: 'string', value: arg1.value };
+    }
+    // sendTo((ctx, e) => actorRef)
+    else if (
+      t.isArrowFunctionExpression(arg1) ||
+      t.isFunctionExpression(arg1)
+    ) {
+      actorRef = {
+        type: 'expression',
+        value: fileContent.slice(arg1.start!, arg1.end!),
+      };
+    }
+
+    // Event
+    // sendTo(, 'eventType')
+    if (t.isStringLiteral(arg2)) {
+      eventObject.type = {
+        type: 'string',
+        value: arg2.value,
+      };
+    }
+
+    // sendTo({type: 'event', id: 1}),
+    else if (t.isObjectExpression(arg2)) {
+      eventObject = extractEventObject(arg2, fileContent);
+    }
+
+    // sendTo((ctx, e) => Record<string, any>)
+    else if (
+      t.isArrowFunctionExpression(arg2) ||
+      t.isFunctionExpression(arg2)
+    ) {
+      eventObject = {
+        type: 'expression',
+        value: fileContent.slice(arg2.start!, arg2.end!),
+      };
+    }
+
+    // Options
+    if (t.isObjectExpression(arg3)) {
+      // delay = string | number | (ctx, e) => string | number
+      const foundDelay = arg3.properties.find(
+        prop =>
+          t.isObjectProperty(prop) &&
+          t.isIdentifier(prop.key) &&
+          prop.key.name === 'delay',
+      ) as t.ObjectProperty | undefined;
+      if (foundDelay) {
+        if (
+          t.isArrowFunctionExpression(foundDelay.value) ||
+          t.isFunctionExpression(foundDelay.value)
+        ) {
+          delay = {
+            type: 'expression',
+            value: fileContent.slice(foundDelay.start!, foundDelay.end!),
+          };
+        } else if (
+          t.isStringLiteral(foundDelay.value) ||
+          t.isNumericLiteral(foundDelay.value)
+        ) {
+          delay = {
+            type: getLiteralType(foundDelay.value),
+            value: foundDelay.value.value,
+          };
+        }
+      }
+
+      // id = string
+      const foundId = arg3.properties.find(
+        prop =>
+          t.isObjectProperty(prop) &&
+          t.isIdentifier(prop.key) &&
+          prop.key.name === 'id',
+      ) as t.ObjectProperty | undefined;
+      if (
+        foundId &&
+        (t.isStringLiteral(foundId.value) || t.isNumericLiteral(foundId.value))
+      ) {
+        id = {
+          type: getLiteralType(foundId.value),
+          value: foundId.value.value,
+        };
+      }
+    }
+
+    return {
+      event: eventObject,
+      actorRef: actorRef!,
+      options: {
+        delay,
+        id,
+      },
+    };
+  }
+
+  throw Error(`Unsupported sendTo expression`);
+}
+
+function getLiteralType(value: t.ObjectProperty['value']) {
+  if (
+    t.isNullLiteral(value) ||
+    t.isRegExpLiteral(value) ||
+    t.isTemplateLiteral(value)
+  ) {
+    throw Error('null,regexp or template literals can not have literal type');
+  }
```

Comment:
what does it mean? I see those being often handled before calling `getLiteralType` and I'm not sure what's the distinction